### PR TITLE
add jemalloc & pkg-config as build deps for Salmon 0.12.0

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
`make test` seems to be hanging with these changes, whereas it runs in just 1 second without including `jemalloc` as a dep, interesting...

**Comment by vanzod** [comment]
@boegel I confirm that the test hangs with `jemalloc`.

**Comment by wpoely86** [comment]
Which version of jemalloc does it ship with?

**Comment by boegel** [comment]
@wpoely86 Same version as the one I'm using here. Which makes it even weirder. Maybe they do something special that doesn't happen if an external `jemalloc` is used. Or maybe it's the `$LD_PRELOAD` we do in `jemalloc`?

**Comment by boegel** [comment]
Unsetting `$LD_PRELOAD` before `make test` fixes the issue for me, but then I wonder what effect it's have on Salmon itself outside of the tests...

**Comment by wpoely86** [comment]
but how does it use jemalloc then?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2402.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/78946bb19ac5946987267379c583f4d8 for a full test report.

**Comment by boegel** [comment]
@wpoely86 Salmon statically links in `jemalloc`, so just a build dep is sufficient (changed in b95f313244).

I've retested the resulting `salmon` binary using the examples explained in https://combine-lab.github.io/salmon/getting_started/, works fine, so this is good to go imho...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2727.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/7b8afabfde988cc5fab120e8b5335314 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3122.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/7462f47e66688b5091eda5914576dfd7 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2006.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/7eff773ebe259a2ff36755826c785620 for a full test report.

File: easybuild/easyconfigs/s/Salmon/Salmon-0.12.0-foss-2018b.eb [hunk 1]
```diff
@@ -15,15 +15,23 @@ source_urls = [GITHUB_LOWER_SOURCE]
...
+]
 
 dependencies = [
     ('Boost', '1.67.0'),
     ('tbb', '2018_U5'),
 ]
 
+configopts = '-DJEMALLOC_ROOT=$EBROOTJEMALLOC'
+
+# need to unset the $LD_PRELOAD set be jemalloc to avoid hanging 'make test'
```
Comment (wpoely86) [inline easybuild/easyconfigs/s/Salmon/Salmon-0.12.0-foss-2018b.eb]: please add that it's statically linked in

Comment (boegel) [reply easybuild/easyconfigs/s/Salmon/Salmon-0.12.0-foss-2018b.eb]: done in 4e085fbabf

**Pull request merged**